### PR TITLE
Slasher proto and function renames

### PR DESCRIPTION
**Pull request by 0xKiwi**
This PR renames and cleans up the slasher further to reduce confusion.

Renames:
* `ValidatorIDToIdxAtt` -> `CompressedIdxAtt`
* `ValidatorIDToIdxAttList` -> `CompressedIdxAttList`
* `BlockHeader()` -> `BlockHeaders()`
* `dataForDetection()` -> `attsAndCommitteesForEpoch()`
* `slasherOldAttestationFeeder()` -> `historicalAttestationFeeder()`

Changed `indexedAttestationsIndicesBucket` to `compressedIdxAttsBucket`

`detectAttestation()` split into `convertToIndexed()`and `detectSlashings()`

Changes the `ClearDB()` function to only clear the slasher db, not the whole DB folder.

### CODE REVIEW

File: slasher/db/indexed_attestations.go [hunk 1]
```diff
@@ -25,8 +25,8 @@ func unmarshalIdxAtt(enc []byte) (*ethpb.IndexedAttestation, error) {
 	return protoIdxAtt, nil
 }
 
-func unmarshalValIDsToIdxAttList(enc []byte) (*slashpb.ValidatorIDToIdxAttList, error) {
-	protoIdxAtt := &slashpb.ValidatorIDToIdxAttList{}
+func unmarshalCompressedIdxAttList(enc []byte) (*slashpb.CompressedIdxAttList, error) {
```
Comment (terencechain) [inline slasher/db/indexed_attestations.go]: CompressedIdx sounds too vague imo, is it validator index? attestation index? block index?... etc

File: slasher/service/data_update.go [hunk 2]
```diff
@@ -9,77 +10,108 @@ import (
...
+				continue
+			}
+			log.Info("detected attestation for target: %d", att.Data.Target)
+		case <-s.context.Done():
+			return status.Error(codes.Canceled, "Stream context canceled")
+		}
+	}
+}
+
+// finalizedChangeUpdater this is a stub for the coming PRs #3133
```
Comment (terencechain) [inline slasher/service/data_update.go]:
```suggestion
// finalizedChangeUpdater this is a stub for the coming PRs #3133.
```

File: slasher/service/data_update.go [hunk 3]
```diff
@@ -9,77 +10,108 @@ import (
...
+// Store validator index to public key map Validate attestation signature.
```
Comment (terencechain) [inline slasher/service/data_update.go]:
```suggestion
// Store validator index to public key map to validate attestation signature.
```

File: slasher/db/indexed_attestations.go (hunk 1 above)
Comment (0xKiwi) [reply slasher/db/indexed_attestations.go]: It is an indexed attestation, but the `AttestationData` is changed to its `[]byte` attestation data root. That's why I refer to it as "compressed".

Comment (terencechain) [reply slasher/db/indexed_attestations.go]: Makes sense 👍

**Pull request merged**